### PR TITLE
IGAPP-197: Resolve Flowfixme in navigator

### PR DESCRIPTION
**Pull request by Taggotty**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

see https://reactnavigation.org/docs/4.x/navigation-actions

### CODE REVIEW

File: native/src/routes/landing/containers/LandingContainer.js [hunk 1]
```diff
@@ -62,22 +62,18 @@ class LandingContainer extends React.Component<ContainerPropsType> {
...
       },
       newKey: key
-    })
-
-    // $FlowFixMe For some reason action is not allowed to be a StackAction
-    navigation.navigate({
-      routeName: 'CityContent',
-      action: action
-    })
+    }))
```
Comment (Schedulaar) [inline native/src/routes/landing/containers/LandingContainer.js:76]: This seems to be wrong... I guess we should dispatch the StackAction in the StackNavigator, not in the top level SwitchNavigator...  (Just an educated guess)

File: native/src/routes/landing/containers/LandingContainer.js [hunk 2]
```diff
@@ -62,22 +62,18 @@ class LandingContainer extends React.Component<ContainerPropsType> {
     const path = `/${cityCode}/${language}`
     const key: string = generateKey()
 
+    navigation.navigate('CityContent')
+
     // $FlowFixMe newKey missing in typedef
-    const action: NavigationReplaceAction = StackActions.replace({
+    navigation.dispatch(StackActions.replace({
```
Comment (maxammann) [inline native/src/routes/landing/containers/LandingContainer.js:68]: What is the difference between dispath and nativate?

**Review by maxammann — APPROVED** [review]

File: native/src/routes/landing/containers/LandingContainer.js (hunk 1 above)
Comment (klinzo) [reply native/src/routes/landing/containers/LandingContainer.js:76]: I think I did this change some time back as well and it led to problems... not sure of which kind though anymore...

File: native/src/routes/landing/containers/LandingContainer.js (hunk 2 above)
Comment (Taggotty) [reply native/src/routes/landing/containers/LandingContainer.js:68]: Dispatch takes an action, while navigate is the function. Furthermore, dispatch can take any Stack/Navigation Action. In our case what we want is described [here](https://reactnavigation.org/docs/nesting-navigators#navigator-specific-methods-are-available-in-the-navigators-nested-inside).

File: native/src/routes/landing/containers/LandingContainer.js (hunk 1 above)
Comment (Taggotty) [reply native/src/routes/landing/containers/LandingContainer.js:76]: It does make sense, but I think it is not possible, as there is no real entry point to a city before you actually enter the city (with specific city and language code). Before you are just somewhere in the switch navigator (in this case in the landing container) which itself is only a basic Navigator.

File: release-notes/unreleased/IGAPP-166-ResolveFlowFixMeInNavigator.yml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+issue_key: IGAPP-166
```
Comment (klinzo) [inline release-notes/unreleased/IGAPP-166-ResolveFlowFixMeInNavigator.yml:1]: imo this should be removed :)

**Review by klinzo — APPROVED** [review]

**Pull request merged**